tests/components/sonos/test_init.py

```diff
@@ -335,11 +370,14 @@ async def test_async_poll_manual_hosts_6(
 
         with caplog.at_level(logging.DEBUG):
             caplog.clear()
-            # Delay for second iteration and other tasks
-            await asyncio.sleep(1.0)
+            # The discovery events should not fire, wait with a timeout.
+            with pytest.raises(asyncio.TimeoutError):
+                await asyncio.wait_for(speaker_1_activity.event.wait(), 1.0)
```

Comment:
`asyncio.wait_for` has some bad cancellation semantics up until cpyton 3.12 so we try not to use it.

Use `async_timeout` for now, we can replace it with `asyncio.timeout` when we drop cpython 3.10 support.


or you can do
```python
if sys.version_info[:2] < (3, 11):
    from async_timeout import timeout as asyncio_timeout
else:
    from asyncio import timeout as asyncio_timeout
```

Reply:
Updated that.  Also fixed something else that was failing in 3.11.  The best news of the day, is this last set of work revealed a bug in the code - which is now corrected.